galaxy/api/v2/views/collection_version.py

```diff
@@ -15,21 +15,70 @@
 # You should have received a copy of the Apache License
 # along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
 
-from rest_framework import exceptions as drf_exc
+from django.shortcuts import get_object_or_404
 from rest_framework import views
-from rest_framework import status as http_codes
+from rest_framework import generics
+from rest_framework.response import Response
 from rest_framework.permissions import IsAuthenticated
 
+from galaxy.main import models
+from galaxy.api.v2 import serializers
+from galaxy.api.v2.pagination import CustomPagination
+
 
 __all__ = (
-    'CollectionVersionView',
+    'VersionListView',
+    'VersionDetailView',
 )
 
 
-class CollectionVersionView(views.APIView):
+# TODO(awcrosby): Move to views/utils.py and reuse across views
+def _lookup_collection(kwargs):
+    """Helper method to get collection from id, or namespace and name."""
+    pk = kwargs.get('pk', None)
+    ns_name = kwargs.get('namespace', None)
+    name = kwargs.get('name', None)
+
+    if pk:
+        return get_object_or_404(models.Collection, pk=pk)
+
+    ns = get_object_or_404(models.Namespace, name=ns_name)
+    return get_object_or_404(models.Collection, namespace=ns, name=name)
+
+
+class VersionListView(generics.ListAPIView):
     permission_classes = (IsAuthenticated, )
```

Comment:
Guessing this is the cause of the auth check / 401 mentioned below.

Does VersionListView / CollectionListView need auth?

Reply:
I pulled the permission_class from the POST, I will update these GET views to the AllowAny permission